scrapy/contrib/loader/processor.py

```diff
@@ -63,6 +63,33 @@ def __call__(self, values):
         return values
 
 
+class JmesProcessor(object):
+    """
+        Query the input string for the jmespath (given at instantiation), and return the answer
+        Requires : jmespath(https://github.com/boto/jmespath), ast
+        Note: JmesProcessor accepts only a single string at a time and returns string/dict/None based on the jmespath query.
+    """
+    def __init__(self, json_path):
+        self.json_path = json_path
+        try:
+            import jmespath
+            self.compiled_path = jmespath.compile(self.json_path)
+        except:
+            raise ImportError("You need to have jmespath(https://github.com/boto/jmespath)"
+                              + " and AST modules before being able to use this processor")
+
+    def __call__(self, value):
+        """Query value for the jmespath query and return answer
+        Input : string
+        Output : string / dict / None
```

Comment:
What about other Python types? 
Could you please also use Sphinx-compatible docstrings, like
```
:param str value: a string with JSON data to extract from
```